### PR TITLE
PM-11484: Add logic to parse URI from AccessibilityNodeInfo

### PR DESCRIPTION
**Pull request by david-livefront**
## 🎟️ Tracking

[PM-11484](https://bitwarden.atlassian.net/browse/PM-11484)

## 📔 Objective

This PR adds the logic for parsing a URI from the `AccessibilityNodeInfo` provided from the `AccessibilityService`.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-11484]: https://bitwarden.atlassian.net/browse/PM-11484?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/model/Browser.kt [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+ * [Browser] if it is a match.
+ */
+fun String.getSupportedBrowser(): Browser? =
+    ACCESSIBILITY_SUPPORTED_BROWSERS.find { it.packageName == this@getSupportedBrowser }
+
+/**
+ * A list of supported browsers and the field ID used to find the url bar.
+ */
+private val ACCESSIBILITY_SUPPORTED_BROWSERS = listOf(
+    Browser(packageName = "alook.browser", urlFieldId = "search_fragment_input_view"),
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/model/Browser.kt:33]: Thoughts on this giant thing?

File: app/src/test/java/com/x8bit/bitwarden/data/autofill/accessibility/parser/AccessibilityParserTest.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        val expectedResult = "androidapp://$testPackageName".toUri()
+
+        val result = accessibilityParser.parseForUriOrPackageName(rootNode = rootNode)
+
+        assertEquals(expectedResult, result)
+    }
+
+    @Suppress("MaxLineLength")
+    @Test
+    fun `parseForUriOrPackageName should return null when package is a supported browser and url bar is not found`() {
```
Comment (brian-livefront) [inline app/src/test/java/com/x8bit/bitwarden/data/autofill/accessibility/parser/AccessibilityParserTest.kt:46]:
```suggestion
    fun `parseForUriOrPackageName should return null when package is a supported browser and URL bar is not found`() {
```

File: app/src/test/java/com/x8bit/bitwarden/data/autofill/accessibility/parser/AccessibilityParserTest.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        val result = accessibilityParser.parseForUriOrPackageName(rootNode = rootNode)
+
+        assertNull(result)
+    }
+
+    @Suppress("MaxLineLength")
+    @Test
+    fun `parseForUriOrPackageName should return the site url as a uri when package is a supported browser and url is found`() {
```
Comment (brian-livefront) [inline app/src/test/java/com/x8bit/bitwarden/data/autofill/accessibility/parser/AccessibilityParserTest.kt:64]:
```suggestion
    fun `parseForUriOrPackageName should return the site URL as a URI when package is a supported browser and URL is found`() {
```

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/model/Browser.kt [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+fun String.getSupportedBrowser(): Browser? =
+    ACCESSIBILITY_SUPPORTED_BROWSERS.find { it.packageName == this@getSupportedBrowser }
+
+/**
+ * A list of supported browsers and the field ID used to find the url bar.
+ *
+ * This list should be kept in order and match the list of compatibility browsers in the
+ * autofill_service_configuration.xml.
+ */
+private val ACCESSIBILITY_SUPPORTED_BROWSERS = listOf(
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/model/Browser.kt:35]: ❓ just for curiosity, are the urlField values sourced from somewhere specific, are we getting those from the existing impl in the MAUI app?

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/model/Browser.kt [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+        possibleUrlFieldIds = listOf(urlFieldId),
+        urlExtractor = urlExtractor,
+    )
+}
+
+/**
+ * Determines if the [String] receiver is a package name for a supported browser and returns that
+ * [Browser] if it is a match.
+ */
+fun String.getSupportedBrowser(): Browser? =
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/model/Browser.kt:26]: ⛏️ would it be in the convention of similar functions like this to suffix with "OrNull"

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/model/Browser.kt (hunk 4 above)
Comment (david-livefront) [reply app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/model/Browser.kt:35]: Yea, these are are pulled from Maui.

**Review by dseverns-livefront — APPROVED** [review]

**Review by brian-livefront — APPROVED** [review]
Looks good!

**Comment by david-livefront** [comment]
Thanks @dseverns-livefront & @brian-livefront

**Pull request merged**